packages/mobile/src/screens/notifications-screen/NotificationListItem.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+  Notification,
+  NotificationType
+} from 'audius-client/src/common/store/notifications/types'
+
+import {
+  FavoriteNotification,
+  FollowNotification,
+  RepostNotification,
+  ChallengeRewardNotification,
+  RemixCreateNotification,
+  UserSubscriptionNotification,
+  RemixCosignNotification,
+  MilestoneNotification,
+  AnnouncementNotification,
+  TierChangeNotification,
+  TrendingTrackNotification
+} from './Notifications'
+
+type NotificationListItemProps = {
+  notification: Notification
+}
+export const NotificationListItem = (props: NotificationListItemProps) => {
```

Comment:
Nit: could be cleaner to add these to a map and index into it! Could even just use the raw type of the React class and instantiate it after pulling it out, since they have the same prop shape

Reply:
id be down! but how do you get typescript to properly type the discriminated type with map access? from what i can tell you need branching logic to line up the correct notif component with the notif prop type?

Reply:
ahh yeah, didn't realize each of these notif types were different shapes and relied on the discrimated union. oh well!